code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -183,47 +183,44 @@
 	to_chat(hiveminde, "<span class='xenonotice'>We can't plant a node without weeds nearby, we've been moved back to our core.</span>")
 	return fail_activate()
 
-// Choose Resin
-/datum/action/xeno_action/choose_resin
-	name = "Choose Resin Structure"
+// Secrete Resin
+/datum/action/xeno_action/activable/secrete_resin
+	name = "Secrete Resin"
 	action_icon_state = "resin wall"
-	mechanics_text = "Selects which structure you will build with the (secrete resin) ability."
-	keybind_signal = COMSIG_XENOABILITY_CHOOSE_RESIN
+	mechanics_text = "Builds whatever resin you selected"
+	ability_name = "secrete resin"
+	plasma_cost = 75
+	keybind_signal = COMSIG_XENOABILITY_SECRETE_RESIN
+	var/base_wait = 1 SECONDS
+	var/scaling_wait = 1 SECONDS
```

Comment:
Autodoc

Reply:
no clue what it does

Reply:
depending of building thing it takes longer to build
```
/datum/action/xeno_action/activable/secrete_resin/proc/get_wait()
	. = base_wait
	if(!scaling_wait)
		return
	var/mob/living/carbon/xenomorph/X = owner
	var/sticky_resin_modifier = 1
	if(X.selected_resin == /obj/effect/alien/resin/sticky) //Sticky resin builds twice as fast
		sticky_resin_modifier = 0.5
	return (base_wait + scaling_wait - max(0, (scaling_wait * X.health / X.maxHealth))) * sticky_resin_modifier
```